subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/report/PackagePageRenderer.java

```diff
@@ -48,7 +48,10 @@ private void renderClasses(SimpleHtmlWriter htmlWriter) throws IOException {
         for (ClassTestResults testClass : getResults().getClasses()) {
             htmlWriter.startElement("tr");
             htmlWriter.startElement("td").attribute("class", testClass.getStatusClass());
-                htmlWriter.startElement("a").attribute("href", asHtmlLinkEncoded(getResults().getUrlTo(testClass))).characters(testClass.getSimpleName()).endElement();
+            htmlWriter.startElement("a")
+                .attribute("href", asHtmlLinkEncoded(getResults().getUrlTo(testClass)))
+                .attribute("title", asHtmlLinkEncoded(testClass.getName()))
```

Comment:
❌ Why is the title link encoded?

Reply:
`<title>` provides a handy way for users to see the original class name (not the display name) if they want - just move mouse cursor to the link. Also, `HtmlTestClassResult` can extract original class name to do some matching work.

Reply:
I think we are misusing the title attribute here. It's not meant as some hidden information, but as a more detailed explanation of what you are clicking on. The name is less detailed than the display name, so it shouldn't be the link title.

Reply:
For the html result scraping you can add a hidden attribute that users won't see.